src/rules/requests/request-rule.ts

```diff
@@ -82,6 +82,12 @@ export class RequestRule implements RequestRule {
                 ])
                 .catch(() => {}) // Ignore handler errors here - we're only tracking the request
                 .then(() => waitForCompletedRequest(req))
+                .catch(() => {
+                    // If for some reason the request is not completed, we still want to record it.
+                    // TODO: Update the body to return the data that has been received so far.
+                    const completedRequest = buildInitiatedRequest(req);
+                    return {...completedRequest, body: buildBodyReader(Buffer.from([]), req.headers) };
```

Comment:
I haven't found yet an elegant way to create a body ready, at this stage, with the data that has been transmitted so far. For now, I created an empty body to solve the promise rejection issue. But it's certainly something that should be improved in the future.

Reply:
Yep, that's fair. I'm not too worried about it to be honest, since it's an unusual edge case and I'm not sure it's worth the complexity, but yes leaving the TODO to look at that in future is fine.